File: client/src/app/modules/miscellaneous/services/paper.set.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { PaperSetType } from '../../miscellaneous/model/PaperSetType';

@Injectable({
  providedIn: 'root'
})
export class PaperSetTypeService extends AppRestService<PaperSetType> {

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getPaperSetTypes() {
    return this.getList(new HttpParams(), "paper-set-types");
  }

  addOrUpdatePaperSetType(paperSetType: PaperSetType) {
    return this.addOrUpdate(new HttpParams(), "paper-set-type", paperSetType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/PaperSetTypeService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.PaperSetType;

public interface PaperSetTypeService {
    public List<PaperSetType> getPaperSetTypes();

    public PaperSetType getPaperSetType(Integer id);
	
	 public PaperSetType addOrUpdatePaperSetType(PaperSetType paperSetType);
}
